src/src.jl

```diff
@@ -10,372 +10,159 @@
+        if norm(support_point - container.center)^2 > container.squared_radius
+            new_center_and_radius!(support_point, container)
+            miniball_support_points(container, support_point)
+            container.fsize -= 1
+            # Move the support points inside the support vector. The idea is to
+            # move the points that are the furthest apart from each other to the
+            # front of the vector. Closest points to gather are at the end of
+            # the vector.
+            splice!(container.support_vector, idx)
+            unshift!(container.support_vector, support_point)
         end
     end
  end
 
-"""
-    pivot_move_to_front(pivot, container)
-
-Appends a pivot point to the front of the support vector.
-"""
-function pivot_move_to_front{F<:AbstractFloat}(pivot::Array{F, 1}, container::MBContainer)
-    dim = container.dim
-    unshift!(container.support_vector, pivot)
-    if length(container.support_vector) == (dim + 2)
-        pop!(container.support_vector)
-    end
- end
-
-"""
-    calculate_miniball(container)
-
-Function for creating the support vector. This function finds the furthest
-point from the current center: the pivot point. The pivot point is added into
-the support vector, which is then used to create the miniball.
-"""
- function calculate_miniball(container::MBContainer)
-     dim   = container.dim
-     len   = container.len
-     arr_squared_r = container.squared_radius
-     fsize = container.fsize
-     support_vector = container.support_vector
-     points = container.points
-     old_sqr_r = container.squared_radius
-     max_distance = 0.0
-     distance = 0.0
-     center = vec(container.center)
-     pivot, max_distance = find_pivot(len, points, center, arr_squared_r, dim)
-     if max_distance > 0.0
+function miniball(points; max_iterations=20)
+     ball = MBContainer(points)
+     fsize = ball.fsize
+     support_vector = ball.support_vector
+     r2_prev = -Inf
+     for i=1:max_iterations
+         center = vec(ball.center)
+         pivot_idx = indmax(norm(points[k,:]-center) for k=1:size(points,1))
+         pivot = points[pivot_idx, :]
          if !(pivot in support_vector)
              @assert fsize == 1
-             if new_center_and_radius!(pivot, container)
-                 miniball_support_points(container, pivot)
-                 container.fsize -= 1
-                 pivot_move_to_front(pivot, container)
+             new_center_and_radius!(pivot, ball)
+             miniball_support_points(ball, pivot)
+             ball.fsize -= 1
+             dim = size(ball.points, 2)
+             unshift!(ball.support_vector, pivot)
+             if length(ball.support_vector) == (dim + 2)
+                 pop!(ball.support_vector)
              end
          end
+         dr2 = ball.squared_radius - r2_prev
+         println("Radius: $(sqrt(ball.squared_radius)), ch: $dr2")
+         isapprox(dr2, 0.0) && break
+         r2_prev = copy(ball.squared_radius)

```

Comment:
`r2_prev = ball.squared_radius` is enough, `Float64` is immutable. This is not a `&double` or something.